src/aiida/tools/pytest_fixtures/daemon.py

```diff
@@ -60,11 +62,61 @@ def test(started_daemon_client):
         daemon_client.start_daemon()
         assert daemon_client.is_daemon_running
 
+    import threading
+    import time
+
+    from watchdog.events import FileSystemEventHandler
+    from watchdog.observers import Observer
+
+    logger = logging.getLogger('tests.daemon:started_daemon_client')
+
+    logger.debug(f'Daemon log file is located at: {daemon_client.daemon_log_file}')
+
+    # This flag will be used to stop the thread when the fixture is torn down
+    stop_thread = False
+
+    class LogFileEventHandler(FileSystemEventHandler):
+        def __init__(self, filepath):
+            self.filepath = filepath
+            # Keep track of how many bytes have been read so we print only new data
+            self._pos = 0
+
+        def on_modified(self, event):
+            if event.src_path == self.filepath:
+                # The file was modified, read from the last known position
+                with open(self.filepath, 'r') as f:
+                    f.seek(self._pos)
+                    new_output = f.read()
+                    if new_output:
+                        logger.debug(new_output)
```

Comment:
Maybe I should not use logger here since the content of the log file is in log format.